src/main/java/com/github/manolo8/darkbot/extensions/features/FeatureRegistry.java

```diff
@@ -65,12 +73,32 @@ public void beforeLoad() {
     @Override
     public void afterLoad() {
         pluginHandler.LOADED_PLUGINS.forEach(pl ->
-                Arrays.stream(pl.getDefinition().features)
+                getFeaturesList(pl)
                         .forEach(feature -> registerPluginFeature(pl, feature)));
         pluginHandler.LOADED_PLUGINS.sort((p1, p2) -> Boolean.compare(isDisabled(p1), isDisabled(p2)));
         registryHandler.update();
     }
 
+    private List<String> getFeaturesList(Plugin plugin) {
+        if (plugin.getDefinition().features.length != 0) return Arrays.asList(plugin.getDefinition().features);
+        URLClassLoader loader = new URLClassLoader(new URL[]{plugin.getJar()});
+        List<String> features = new LinkedList<>();
```

Comment:
```suggestion
        List<String> features = new ArrayList<>();
```
just use an array list, it's not like a linked list provides any benefit here